src/ui/menu.py

```diff
@@ -1,309 +1,231 @@
+        print(f'\nAttempting to Remove {len(cards)} Cards...\n\n')
+        count = 0
+        for card in cards:
+            variant = card[1]
+            print(
+                'MID:', card[0]['multiverse_ids'], ' ',
+                'Name:', card[0]['name'], ' ',
+                'Mana:', card[0]['mana_cost'], ' ',
+                'Type:', card[0]['type_line'], ' ',
+                'Set:', card[0]['set_name'], ' ',
+                'Rarity:', card[0]['rarity'], ' '
+                'Variant:', variant, ' '
+                )
+            try:
+                inventory.removeCardFromInventory(card[0], variant)
+                count += 1
+            except Exception as e:
+                print(e)
+                placeholder = input('')
+                continue
+        print('\nCards removed.')
+        placeholder = input('Press any key to continue...\n')
+
+    # SEARCH MENU FUNCTIONS
+    # TODO: Implement search by block functionality in db
+    def __searchBlock(self, params):
+        print('\n' * 50)
+        blockname = input('Enter name of block to search by\n\n>')
+        print('\n\nSearching...')
+        placeholder = input('Press any key to continue...\n')
 
-def __removeCard(inventory):
-    print('\n'*30)
-    print('*** Card Remove ***')
-    print('Is this card a token or emblem? [Y/N]')
-    reply = input('\n>')
-    if reply.lower() == 'y':
-        print('Please enter the name of the card.')
-        name = input('\n>')
-        cards = inventory.searchByName(name)
-        card = menu_utils.singleSearchResponseEval(cards)
-        if card:
-            print('Removing card from inventory')
-            inventory.removeCard(card)
+    def __searchSet(self, params):
+        print('\n' * 50)
+        setname = input('Enter name of set to search by\n\n>')
+        if isinstance(params, list):
+            inventory = params[0]
+            action = params[1]
         else:
-            print('\n\nReturning to menu')
-            placeholder = input('Press any key to continue...\n>>')
-    elif reply.lower() == 'n':
-        print('Please enter ID of the card [Front Face ID if double sided].')
-        id = input('\n>')
-        cards = inventory.searchByMID(id)
-        card = menu_utils.singleSearchResponseEval(cards)
-        if card:
-            print('Removing card from inventory')
-            inventory.removeCard(card)
+            inventory = params
+            action = None
+
+        try:
+            set = inventory.searchBySet(setname)
+            selected_cards = self.cache.createFromDBResponse(set, setname, action)
+            if selected_cards:
+                if action == 'add':
+                    self.__addCards(inventory, selected_cards)
+                elif action == 'remove':
+                    self.__removeCards(inventory, selected_cards)
+        except Exception as e:
+            print(e)
+            placeholder = input('')
+            
+    def __searchName(self, params):
```

Comment:
For `__addCards`, this `__searchName` function will search the whole listing of all 60k possible cards. --that is good we could add any one of them theoretically.

For `__removeCards`, it searches the same listing of 60k cards, and prompts the user to select one of those, even if the card is not actually in stock in the users database/inventory.

Can we make it so the `__removeCards` only passes a lists of cards that have a foil/nonfoil > 0?

Reply:
Will be implemented in another branch.